### PR TITLE
plasma rifle

### PR DESCRIPTION
**Pull request by yglop**
## About the PR
Plasma rifle (has great damage and may explode with 1/6 chance) ((can be researched and printed at secfab)).

## Why / Balance
Its funny.

## Technical details
Added WeaponRandomExplode component and system.

## Media
plasma_rifle.png

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- add: Plasma rifle.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new weapon, the Plasma Rifle, with a 16% chance of explosion upon firing.
	- Added a random explosion mechanic for weapons, enhancing gameplay unpredictability.
	- New projectile type, BulletPlasmaCutterStrong, featuring increased damage and visibility effects.
	- Added crafting recipes for the Plasma Cutter and Plasma Rifle.
	- New localized term for the Plasma Rifle technology.

- **Gameplay Enhancements**
	- Expanded weapon whitelist for material storage.
	- Introduced a new technology for researching the Plasma Rifle within the arsenal system.

### CODE REVIEW

**Review by Piras314 — CHANGES_REQUESTED** [review]
nerf it down to the ground (like SIGNIFICANTLY) and make it cheap to make, 40 heat damage 400 structural damage shooting twice per second is RIDICULOUS even with the whole blowing up thing

**Comment by yglop** [comment]
@Piras314 what a boring man you are (i nerfed it)

**Review by Piras314 — CHANGES_REQUESTED** [review]
that is more structural damage than a throngler does per hit, and 30 heat damage instantly is wild, decrease the damage a lot, increase the fire rate a bit

**Comment by yglop** [comment]
@Piras314 its not hitscan. laser cannon has 28 HITSCAN heat damage. also 200 structural will destroy a normal wall it 4 hits. and also you forgot that the rifle MAY EXPLODE AND KILL THE USER WITH 1D6 DICE ROLL.

**Comment by Piras314** [comment]
the fact that it explodes every 6 times means that it should absolutely be a crappy weapon, having a really good weapon be so shottily made that it just blows up all the time is ridiculous

**Comment by JoeHammad1844** [comment]
handmade laser musket?

**Comment by a-person5660** [comment]
> the fact that it explodes every 6 times means that it should absolutely be a crappy weapon, having a really good weapon be so shoddily made that it just blows up all the time is ridiculous

imo this should have a safe mode where its more equivilent to other laser weapons and an unsafe mode that functions similarly to the way it does here. this also has taken inspiration from the WH40k plasma pistol which has a chance to explode like this.

i dont see any issue with high cost, high damage weapons like this having a silly trade off, i can imagine some nukie fights where a clutch or fail would be made by this and it would almost always be funny.
making this weapon low cost, low damage and still explode would simply mean it would never be used.

File: Resources/Locale/en-US/Goobstation/research/technologies.ftl [hunk 1]
```diff
@@ -0,0 +1 @@
+research-technology-weapon-plasma-rifle = Experemental plasma rifle
```
[the remark this answers is not in the document]
Comment (yglop) [reply Resources/Locale/en-US/Goobstation/research/technologies.ftl]: fuck you. thanks.

File: Content.Server/Goobstation/WeaponRandomExplode/WeaponRandomExplodeComponent.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+namespace Content.Server.Goobstation.WeaponRandomExplode;
+
+[RegisterComponent]
+public sealed partial class WeaponRandomExplodeComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public float explosionChance;
+
+    [DataField, AutoNetworkedField]
+    public float multiplyByCharge;
```
Comment (whateverusername0) [inline Content.Server/Goobstation/WeaponRandomExplode/WeaponRandomExplodeComponent.cs:10]: please clarify what does this do

**Review by whateverusername0 — CHANGES_REQUESTED** [review]
all good just leave some comments

**Comment by Piras314** [comment]
also fix the merge conflicts

**Review by whateverusername0 — APPROVED** [review]
wohks

**Review by Piras314 — APPROVED** [review]

**Comment by yglop** [comment]
> @yglop needs changelog

Forgot to uncomment it.

**Pull request merged**